app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -346,14 +346,25 @@ export default function createRPCMethodTrackingMiddleware({
               ];
             }
           } else if (method === MESSAGE_TYPE.ETH_SIGN_TYPED_DATA_V4) {
-            const { primaryType } = parseTypedDataMessage(data);
-            eventProperties.eip712_primary_type = primaryType;
-            if (PRIMARY_TYPES_PERMIT.includes(primaryType)) {
+            const parsedMessageData = parseTypedDataMessage(data);
+            sensitiveEventProperties = {};
+
+            eventProperties.eip712_primary_type = parsedMessageData.primaryType;
+            sensitiveEventProperties.eip712_verifyingContract =
+              parsedMessageData.domain.verifyingContract;
+            sensitiveEventProperties.eip712_domain_version =
+              parsedMessageData.domain.version;
+            sensitiveEventProperties.eip712_domain_name =
+              parsedMessageData.domain.name;
```

Comment:
could do something like this. Would need to update primaryType again if this suggestion is accepted 

```suggestion
            const { domain, primaryType } = parseTypedDataMessage(data);

            sensitiveEventProperties = {
              eip712_verifyingContract: domain.verifyingContract,
              eip712_domain_version: domain.version,
              eip712_domain_name: domain.name,
            };

            eventProperties.eip712_primary_type = primaryType;
```